lib/util/src/rest_dt.c

```diff
@@ -125,15 +126,16 @@ rest_dt_get(
     if (!buf)
         return merr(ENOMEM);
 
-    sprintf(buf, "/%s", path);
+    snprintf(buf, bufsz, "/%s", path);
+    strncpy(dt_path, buf, sizeof(dt_path));
```

Comment:
Use strlcpy() and check for string truncation for both?

Reply:
yes, I'll fix that.

Reply:
Hmm, rest_dt_put() (above) declares a 1M buffer on the stack, that has got to made dynamic..

Reply:
I think the RFC require supporting a URI of 8000 octets, but most browsers have a limit around 2000.  Common wisdom is that any URI longer that 2000 is broken...

https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers

Reply:
Also, doesn't MHD have a define somewhere for max URI length?  If so, I would use that rather than PATH_MAX, since the two are not of the same ilk...

Reply:
I am unfamiliar with the MHD public API.

Reply:
Yeah, I didn't find that either. So I'm sticking to PATH_MAX.
@beckerg I've changed rest_dt_put to allocate memory for _buf_ on the heap.